### PR TITLE
Port v8 delivery_carrier_label_gls

### PR DESCRIPTION
**Pull request by bealdav**
Needs review

### CODE REVIEW

File: delivery_carrier_label_gls/README.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+================================
+Delivery Carrier Label GLS Group
+================================
+
+
+GLS Group - Transportation services
++++++++++++++++++++++++++++++++++++
+
+Unibox service implementation:
+- send delivery order informations and parcel infothe GLS unibox server
```
Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst]: add an empty line before the list

Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst]: a/infothe/info to the

File: delivery_carrier_label_gls/README.rst [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+- webservice routing info in back
```
Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst]: in background

File: delivery_carrier_label_gls/README.rst [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+- label GLS Unibox generate
```
Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst]: generation

File: delivery_carrier_label_gls/README.rst [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+
+GLS carrier https://gls-group.eu/
+
+|
```
Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst:22]: remove

File: delivery_carrier_label_gls/README.rst [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+* Go to the menu Settings > Configuration > Warehouse, and check 'Use packages: pallets, boxes'
```
Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst]: Better use enumerated list

File: delivery_carrier_label_gls/README.rst [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+* Go to the menu Settings > Configuration > Carriers > GLS
+* Complete account parameters
+* Complete company country in Settings > Companies
+* Install 'Document' module
```
Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst]: This should be in the installation part

File: delivery_carrier_label_gls/README.rst [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+.. image:: /delivery_carrier_label_gls/static/description/gls1.png
+   :alt: Account GLS settings by company Odoo ERP
+
+|
```
Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst:37]: remove

File: delivery_carrier_label_gls/README.rst [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+Usage
+=====
+
+To use this module, you need to create a Delivery Order with Carrier field 'GLS Group'
+
+Odoo Delivery Order with GLS carrier
+++++++++++++++++++++++++++++++++++++
+
+.. image:: /delivery_carrier_label_gls/static/description/gls2.png
+   :alt: Odoo Delivery Order avec GLS carrier with Odoo ERP
```
Comment (elicoidal) [inline delivery_carrier_label_gls/README.rst:48]: suggestion: add the following to all images
:width:800

File: delivery_carrier_label_gls/models/company.py [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    gls_inter_contact_id = fields.Char(
+        string='International',
+        size=10,
+        help='Contact id for GLS International transportation (T8914)')
+    gls_traceability = fields.Boolean(
+        string='Traceability',
+        help="Record traceability informations in Delivery Order "
+             "attachment: web service request and response")
+    gls_generate_label = fields.Boolean(
+        string='Generate Label Automatically',
```
Comment (elicoidal) [inline delivery_carrier_label_gls/models/company.py]: Automatically Generate Label

File: delivery_carrier_label_gls/models/config.py [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        return self.env.user.company_id
+
+    company_id = fields.Many2one(
+        comodel_name='res.company', string='Company',
+        required=True, default=_default_company)
+    customer_code = fields.Char(
+        string='Customer Code',
+        readonly=True, oldname='gls_customer_code',
+        help="Code for GLS carrier company (T8915)\n"
+             "Information common to whole companies "
```
Comment (elicoidal) [inline delivery_carrier_label_gls/models/config.py]: s/whole/all the

File: delivery_carrier_label_gls/models/config.py [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+             "to configure in System Parameter")
```
Comment (elicoidal) [inline delivery_carrier_label_gls/models/config.py]: to be configured in System Parameters.

File: delivery_carrier_label_gls/models/config.py [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+    warehouse = fields.Char(
+        string='Warehouse',
+        readonly=True, oldname='gls_warehouse',
+        help="GLS warehouse near customer location (T8700)\n"
+             "Information common to whole companies "
+             "to configure in System Parameter")
```
Comment (elicoidal) [inline delivery_carrier_label_gls/models/config.py]: see above

File: delivery_carrier_label_gls/models/config.py [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+        param_m = self.env['ir.config_parameter']
+        for field in ['customer_code', 'warehouse']:
+            if field in fields:
+                configs = param_m.search(
+                    [('key', '=', 'carrier_gls_%s' % field)])
+                if not configs:
+                    raise UserError(
+                        "Missing parameter",
+                        "'%s' key is missing in 'System Parameter':\n"
+                        "Add it and set the corresponding value" % field)
```
Comment (elicoidal) [inline delivery_carrier_label_gls/models/config.py]: add a period at the end

File: delivery_carrier_label_gls/models/stock.py [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+                _("Cannot open tracking URL for this carrier "
+                  "because this package "
+                  "doesn't have a tracking number."))
+        return {
+            'type': 'ir.actions.act_url',
+            'url': self._get_carrier_tracking_url(),
+            'target': 'new',
+        }
+
+#
```
Comment (elicoidal) [inline delivery_carrier_label_gls/models/stock.py]: Please remove dead code

File: delivery_carrier_label_gls/models/stock.py [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+
+    @api.multi
+    def do_detailed_transfer(self):
+        for transfert in self:
+            for item in transfert.item_ids:
+                carrier_type = transfert.picking_id.carrier_type
+                if carrier_type and carrier_type == 'gls':
+                    if not (item.package_id or item.result_package_id):
+                        raise UserError(
+                            u"Pour le transporteur '%s' \ntous les produits "
```
Comment (elicoidal) [inline delivery_carrier_label_gls/models/stock.py]: Please translate the message

**Comment by elicoidal** [comment]
@bealdav thanks for the contribution!

File: delivery_carrier_label_gls/README.rst (hunk 4 above)
Comment (bealdav) [inline delivery_carrier_label_gls/README.rst:22]: it's a break line, why remove ?

**Comment by yvaucher** [comment]
There is an error in the tests:

```
2016-06-22 12:56:21,035 24113 ERROR openerp_test openerp.addons.delivery_carrier_label_gls.tests.test_gls: ` openerp.exceptions.Warning: Min size for field 'outbound_depot' is 6 :  4 given
```

**Comment by bealdav** [comment]
Error with runbot `AttributeError: 'Country' object has no attribute 'alpha_2'`
Runbot not use the right version of pycountry (16.11.8) defined in travis.yml
I don't know why ?

**Comment by bealdav** [comment]
https://pypi.python.org/pypi/pycountry changes doc say:

> Attribute names are more coherent now, too. Note that “alpha2”, “alpha4”, etc. are now using an underscore as that’s the pattern in the upstream packages. So it’s “alpha_2” now.

Then I switch my code to `alpha_2`

@gurneyalex could you tell me how ci/runbot manage versions of libs.
Does it uses travis.yml ? requirement.txt ?
How can we explain the wrong version taken account (with alpha2) ?

**Comment by bealdav** [comment]
@pedrobaeza you're also impacted there
https://github.com/OCA/community-data-files/blob/8.0/base_iso3166/models/res_country.py#L30-L35
FYI,  your module doesn't work with the new version of the lib.

**Comment by bealdav** [comment]
any news @gurneyalex @yvaucher ?

**Review by yvaucher — APPROVED** [review]

**Comment by gurneyalex** [comment]
@bealdav @yvaucher @pedrobaeza 
This is a good test. I've enabled docker build on this repository, and triggered a rebuild for this PR. Let's see if this fixes the issue.

**Comment by gurneyalex** [comment]
runbot is red (actually orange) because

2016-11-16 09:33:06,780 133 INFO openerp_test openerp.modules.loading: loading delivery_carrier_label_gls/demo/product.xml
2016-11-16 09:33:07,194 133 WARNING openerp_test openerp.models: stock.inventory.line.create() with unknown fields: product_uom
2016-11-16 09:33:07,220 133 WARNING openerp_test openerp.models: stock.inventory.line.create() with unknown fields: product_uom

**Comment by bealdav** [comment]
@gurneyalex initialisation failed

**Comment by gurneyalex** [comment]
@bealdav sorry?

**Comment by yvaucher** [comment]
stock.inventory.line.create() with unknown fields: product_uom -> product_uom_id

https://github.com/OCA/OCB/blob/8.0/addons/stock/stock.py#L2912

**Comment by bealdav** [comment]
Thanks I'll push

**Review by alexis-via — APPROVED** [review]

File: delivery_carrier_label_gls/models/config.py [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+        string='Customer Code',
+        readonly=True, oldname='gls_customer_code',
+        help="Code for GLS carrier company (T8915)\n"
+             "Information common to all the companies "
+             "to be configured in System Parameter")
+    warehouse = fields.Char(
+        string='Warehouse',
+        readonly=True, oldname='gls_warehouse',
+        help="GLS warehouse near customer location (T8700)\n"
+             "Information common to whole companies "
```
Comment (hparfr) [inline delivery_carrier_label_gls/models/config.py]: setting for all companies

File: delivery_carrier_label_gls/models/company.py [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+    gls_traceability = fields.Boolean(
+        string='Traceability',
+        help="Record traceability informations in Delivery Order "
+             "attachment: web service request and response")
+    gls_generate_label = fields.Boolean(
+        string='Automatically Generate Label',
+        help="Generate label when delivery is done")
+    gls_test = fields.Boolean(
+        string='Url Test',
+        help="Check if requested webservice is test plateform")
```
Comment (hparfr) [inline delivery_carrier_label_gls/models/company.py]: Use testing webservice

File: delivery_carrier_label_gls/models/stock.py [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+
+class StockQuantPackage(models.Model):
+    _inherit = 'stock.quant.package'
+
+    carrier_id = fields.Many2one('delivery.carrier')
+
+    @api.multi
+    def _get_carrier_tracking_url(self):
+        if self.carrier_id.type == 'gls':
+            res = ("https://gls-group.eu/FR/fr/suivi-colis?"
```
Comment (hparfr) [inline delivery_carrier_label_gls/models/stock.py]: use a const

**Review by hparfr — APPROVED** [review]
Minor remarks

**Comment by hparfr** [comment]
:+1:

**Comment by sebastienbeau** [comment]
:+1:  ok for me too

**Comment by bealdav** [comment]
please @codingforfun could you review my PR above ?

**Comment by bealdav** [comment]
@codingforfun sorry for the inconvenient. 
Now fixed.

**Comment by codingforfun** [comment]
THX a lot. 
Sorry I hadn't the time yet for review.

**Pull request merged**